lib/crash_course_agent.ex

```diff
@@ -0,0 +1,35 @@
+defmodule CrashCourseAgent do
+  use GenServer
+  @moduledoc """
+  Gen Server with Agent API
+  """
+  # Client
+  def start_link(state \\ %{}) do
```

Comment:
`def start_link(state \\ %{}) do` is a wrong signature, leading to issues in some cases. It must differ. Please try to figure out why :)